Lib/test/test_typing.py

```diff
@@ -3381,6 +3381,20 @@ class C(Generic[T]): pass
         with self.assertRaises(TypeError):
             C[()]
 
+    def test_generic_subclass_checks(self):
```

Comment:
I think a better place for these tests would be in `test_genericalias.py`, next to this method:

https://github.com/python/cpython/blob/b22d021ee6309f9b0d43e683056db84cfb6fd3c0/Lib/test/test_genericalias.py#L337

Reply:
I placed them here to be sure that `types.GenericAlias` and `typing._GenericAlias` do the same thing. It would be hard to do in other modules.

Reply:
That's reasonable. I don't have a strong opinion either way; let's see what other people think :)